autocards.py

```diff
@@ -1,63 +1,285 @@
+        title.strip()
+        if title == "":
+            print(f"Couldn't find title of the page")
+            title = source
+
+        valid_sections = []  # remove text sections that are too short:
+        for section in el:
+            section = ' '.join(section.get_text().split())
+            if len(section) > 40:
+                valid_sections += [section]
+            else:
+                print(f"Ignored string because too short: {section}")
+
+        if not valid_sections:
+            print("No valid sections found, change the 'element' argument\
+ to look for other html sections than 'p'")
+            return None
+
+        for section in tqdm(valid_sections,
+                            desc="Processing by section",
+                            unit="section"):
+            section = self._sanitize_text(section)
+            self._call_qg(section, title)
+
+    def clear_qa(self):
+        "Delete currently stored qa pairs"
         self.qa_pairs = []
+        global n, cur_n
+        n = 0
+        cur_n = n
 
-    def print(self, prefix='', jeopardy=False):
-        if prefix != '':
+    def string_output(self, prefix='', jeopardy=False):
+        "Return qa pairs to the user"
+        if prefix != "" and prefix[-1] != ' ':
             prefix += ' '
-            
+        if len(self.qa_pairs) == 0:
+            print("No qa generated yet!")
+            return None
+
+        res = []
         for qa_pair in self.qa_pairs:
             if jeopardy:
-                print('\"' + prefix + qa_pair['answer'] + '\",\"' + qa_pair['question'] + '\"')
+                string = f"\"{prefix}{qa_pair['answer']}\",\"\
+{qa_pair['question']}\""
             else:
-                print('\"' + prefix + qa_pair['question'] + '\",\"' + qa_pair['answer'] + '\"')
+                string = f"\"{prefix}{qa_pair['question']}\",\"\
+{qa_pair['answer']}\""
+            res.append(string)
+        return res
 
-    def export(self, filename, prefix='', jeopardy=False):
-        if prefix != '':
+    def print(self, *args, **kwargs):
+        "Print qa pairs to the user"
+        print(self.string_output(*args, **kwargs))
+
+    def pprint(self, *args, **kwargs):
+        "Prettyprint qa pairs to the user"
+        pprint(self.string_output(*args, **kwargs))
+
+    def pandas_output(self, prefix=''):
+        if len(self.qa_pairs) == 0:
+            print("No qa generated yet!")
+            return None
+        "Output a Pandas DataFrame containing qa pairs and metadata"
+        df = pd.DataFrame(columns=list(self.qa_pairs[0].keys()))
+        for qa in self.qa_pairs:
+            df = df.append(qa, ignore_index=True)
+        for i in df.index:
+            for c in df.columns:
+                if pd.isna(df.loc[i, c]):
+                    # otherwise export functions break:
+                    df.loc[i, c] = "Error"
+        return df
+
+    def csv_export(self, filename, prefix='', jeopardy=False):
```

Comment:
Pandas and other modules seem to use the `to_csv` pattern, and I think it's familiar to many. So we could maybe do `to_csv()`, `to_json()`, `to_df()`, etc.

Reply:
done, but I will need to change also the PR of the website then

Reply:
done the commit of the website!